### PR TITLE
Support hashed host identifiers in known_hosts

### PR DESCRIPTION
**Pull request by kangtastic**
# Description
Machines with `HashKnownHosts yes` in their ssh_config store host identifiers (i.e. an IP address or a DNS-resolvable name) for new entries in known_hosts in a hashed form, breaking strict host key checking as currently implemented. Entries with hashed host identifiers may coexist in the same known_hosts file with unhashed host identifiers.

Supporting such entries is possible without any external dependencies, as the hash is a standard RFC 2104-compliant HMAC-SHA1 of the host identifier and the formatted hash that ends up in known_hosts includes the salt/secret key.

- Add a hashed host identifier-aware lookup method to the SSHKnownHosts class
- Modify transport plugins that do strict host key checking to use this method
- Minor docstring corrections

Remedies #174.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

Just running `make docs` again is probably all the "documentation update" needed.

# How Has This Been Tested?

A couple of basic pytests are included, limited to verifying that a SSHKnownHosts object contains a dict with expected key names, but this is sufficient to verify that a hashed host identifier is properly computed and matched.

Functional testing was done using a debugger while connecting to a switch whose hashed host identifier was present in the local known_hosts file with strict host verification turned on. As expected, it worked.

# Checklist:

- [x] My code follows the style guidelines of this project (no GitHub actions complaints! run `make lint` before
 committing!)
- [x] I have commented my code, pydocstyle and darglint are happy, docstrings are in google docstring format, and all
 docstrings include a summary, args, returns and raises fields (even if N/A)
- [x] I have added tests that prove my fix is effective or that my feature works, if adding "functional" tests please
 note if there are any considerations for the vrnetlab setup
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: scrapli/ssh_config.py [hunk 1]
```diff
@@ -445,13 +447,13 @@ def __init__(self, ssh_known_hosts_file: str) -> None:
 
     def _parse(self) -> Dict[str, Dict[str, str]]:
         """
```
Comment (kangtastic) [inline scrapli/ssh_config.py:449]: The original docstring here appears to have been copied from SSHConfig._parse() above and never changed, so let's bring it in line with the other comments in the SSHKnownHosts class.

File: scrapli/ssh_config.py [hunk 2]
```diff
@@ -475,6 +477,36 @@ def _parse(self) -> Dict[str, Dict[str, str]]:
 
         return known_hosts
 
+    def lookup(self, host: str) -> Dict[str, str]:
```
Comment (kangtastic) [inline scrapli/ssh_config.py:480]: This method is modeled on SSHConfig.lookup() above in more ways than one.

**Comment by kangtastic** [comment]
I see codecov is being a silly. 😄 But I did use a 3.7+ only API function in a 3.6 project, and that's fixed now!

Anyway, it should go without saying that I'm totally open to explaining or changing things - just let me know.

File: scrapli/ssh_config.py (hunk 1 above)
Comment (carlmontanari) [reply scrapli/ssh_config.py:449]: heh, copy/paste -- sounds like me 😁 good catch!

**Comment by carlmontanari** [comment]
Thank you so much for this awesome addition @kangtastic ! 

LGTM, will merge and close that other issue. Again, thanks for the great work!

Carl

**Pull request merged**